src/lottie/lottieparser.cpp

```diff
@@ -526,7 +533,7 @@ const char *LottieParserImpl::GetString()
 {
     if (st_ != kHasString) {
         st_ = kError;
-        RAPIDJSON_ASSERT(false);
+        Validate(false);
         return nullptr;
```

Comment:
I noticed this nullptr may be passed to std::string somewhere. That is not allowed and may lead to an undefined behavior.

Reply:
yes you are right . we can remove some of those validate functions . regarding nullptr , as we are going to throw away the parsing data as we already set the error state to null. I just kept it that way without much change.

-- Second, Validate(PeekType() == ??) lines followed by GetString(), GetInt(), EnterObject() , or similar are also unneeded 
    because the functions check the next type.
   >> correct , we can get rid of the most of the validate() code just by putting the parser in error state when reading the value. will update the patch and let you know.